fesvr/htif.cc

```diff
@@ -117,6 +117,8 @@ std::map<std::string, uint64_t> htif_t::load_payload(const std::string& payload,
   if (path.empty())
     throw std::runtime_error(
         "could not open " + payload +
+        " The search path is the current directory, followed by the 'bin' directory within the directory specified by the configure —with-target"+
+        " within the directory specified by the configure —prefix(-prefix/-with-tartget/bin/<target program>)."+
```

Comment:
This line is confusing.

Reply:
Yes, it's indeed confusing and there must be a better way to express, but I think it should be here. The search path is generated by both **--prefix** and **--with-target**. 

For example, one type `../configure --prefix=/usr/local --with-target=/riscv-linux-gnu` when he built spike. He type `spike pk`, spike will try to run `access("pk", F_OK)`. If it can't access, spike will generate a absolute path /usr/local/riscv-linux-gnu/bin/pk(--prefix/--with-target/bin/\<target program\>) and try to run `access("/usr/local/riscv-linux-gnu/bin/pk", F_OK)`. If it false again, the runtime_error will be thrown.

This is why `spike /usr/local/bin/pk` always work, while `spike pk` don't. The absolute path spike generated may be wrong with default --prefix(/usr/local) and default --with-target(riscv64-unknown-elf). If we mention how search path is generated, it will be easier to figure out what's wrong.

Maybe we could mention it in README?

Reply:
> The search path is the current directory, followed by the 'bin' directory within the directory specified by the configure —prefix and —with-target. (--prefix/--with-tartget/bin/<target program>)
Is it better?

Reply:
I agree the error message given in this case could be improved, though the new error message here is quite verbose and complex. How about something like this:
```
Error: could not open pk; searched paths:
  . (current directory)
  /usr/local/riscv-linux-gnu/bin/ (based on configured --prefix and --with-target)
```

Reply:
I pull a new commit with @scottj97 's suggestion and add a `else` below the second `if(access(test_path,F_OK) == 0)` which will throw the error with searched path.

e.g.
`spike ls` will get:

```
terminate called after throwing an instance of 'std::runtime_error'
  what():  could not open ls; searched paths:
        . (current directory)
        /usr/local/riscv64-linux-gnu/bin/ (based on configured --prefix and --with-target)
```
things like `spike /tmp/ls` will still get:
```
terminate called after throwing an instance of 'std::runtime_error'
  what():  could not open /tmp/ls (did you misspell it? If VCS, did you forget +permissive/+permissive-off?)
```